src/proxy.ts

```diff
@@ -0,0 +1,110 @@
+import { fork, Sequence } from 'effection';
+import { on } from '@effection/events';
+import * as proxy from 'http-proxy';
+import * as http from 'http';
+import { listen, ReadyCallback } from './http';
+import { forkOnEvent,  } from './util';
+import * as trumpet from 'trumpet';
+import * as zlib from 'zlib';
+import { Readable, Writable } from 'stream';
+
+interface ProxyOptions {
+  port: number;
+  targetPort: number;
+  inject?: string;
+};
+
+function pipe(from, to) {
+  fork(function*() {
+    let writer = forkOnEvent(from, 'data', function*(chunk) {
```

Comment:
Most of my questions on this are on how streaming is handled. For example, are we sure that streaming errors will be propagated? I haven't work with node streams enough to know for sure. Also, there are events like `finish` and `end` which I'm not sure what the difference is.

Reply:
It seems that streams emit an error even if they have problems. Also there's a neat `pipe` method which makes it super easy to forward these things. I've taken an approach of using the `pipe` method and also attaching error monitors. Seems to work okay, I think.